presubmit.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+# Copyright 2023 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Internal global flags for enabling checks (1 for true, 0 for false)
+ENABLE_FMT=1
+ENABLE_TEST=1
+ENABLE_VET=1
+ENABLE_LINT=1
+ENABLE_STATICCHECK=1
+ENABLE_BUILD=1
+
+# Stop on error flag (1 for true, 0 for false)
+STOP_ON_ERROR=0
+
+RESULTS_DIR="./health_check_results"
+
+# Exit on unset variable
+set -u
+
+error_trap() {
+  echo "An error occurred in script execution. Check the output above for more details."
+  (( STOP_ON_ERROR )) && exit 1
+}
+
+trap error_trap ERR
```

Comment:
nit: The same effect can be achieved with just `set -e`. The message is redundant in my opinion.
```
if [[ "$STOP_ON_ERROR" -ne 0 ]]; then
    set -e
fi
```

Reply:
Done